pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -1369,3 +1369,82 @@ def bubble_sort(array, **kwargs):
         array._modify(force=True)
 
     return array
+
+
+# Under evaluation currently
+def radix_sort(array, **kwargs):
+    """
+        Performs Radix Sort on given input array
+        Current limitations: Works for integers >= 0
+        Parameters
+        ==========
+
+        array: Array
+            The array which is to be sorted.
+
+        backend: pydatastructs.Backend
+            The backend to be used.
+            Optional, by default, the best available
+            backend is used.
+
+        Returns
+        =======
+
+        clean_array: Array
+            The sorted array devoid of null values
+
+        Examples
+        ========
+        >>> from pydatastructs import DynamicOneDimensionalArray as DODA, bucket_sort
+        >>> arr = DODA(int, [2,235,344,14,5000,12,44,73,91,33,8,0])
+        >>> print(*(radix_sort(arr)))
+    """
+    raise_if_backend_is_not_python(
+        radix_sort, kwargs.get('backend', Backend.PYTHON))
+
+    # def clean(x):
+    # Auxiliary function to remove null values, fails while filtering when returns 0(considered False)
+    #     if x == 0:
+    #         return int(0)
+    #     if x is not None:
+    #         return x
+    # clean_array = list(filter(clean, array))
+
+    clean_array = list()
+    for i in array:
+        if i is not None:
+            clean_array.append(i)
```

Comment:
Do not do this. Extra overhead not of much use. `._modify` call below will shift the `None` towards the end which can be replaced by `append` calls in future.

Reply:
Thanks for the suggestion, I have removed the `clean_array` system from the algorithm